sdk/tests/invariant.test.ts

```diff
@@ -248,4 +249,132 @@ describe('invariant', async () => {
     const pools = await invariant.getPools(account)
     assert.deepEqual(pools.length, 1)
   })
+
+  it('should withdraw protocol fee', async () => {
+    if (!token0.contract || !token1.contract || !invariant.contract) {
+      return
+    }
+
+    const feeTier = newFeeTier({ v: 10000000000n }, 1)
+
+    await invariant.addFeeTier(account, feeTier)
+
+    await invariant.createPool(
+      account,
+      token0.contract.address.toString(),
+      token1.contract.address.toString(),
+      feeTier,
+      { v: 1000000000000000000000000n },
+      0n
+    )
+
+    await token0.approve(account, invariant.contract.address.toString(), 10000000000000n)
+    await token1.approve(account, invariant.contract.address.toString(), 10000000000000n)
+
+    const poolKey = newPoolKey(
+      token0.contract.address.toString(),
+      token1.contract.address.toString(),
+      feeTier
+    )
+
+    await invariant.createPosition(
+      account,
+      poolKey,
+      -10n,
+      10n,
+      { v: 10000000000000n },
+      { v: 0n },
```

Comment:
Abusing slippage tolerance can generate undetected errors in tests. Therefore, we pass the current sqrt price on the pool. You can check it on master also